Client/Coordinators/Library/BookmarksCoordinator.swift

```diff
@@ -0,0 +1,51 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Storage
+
+protocol BookmarksCoordinatorDelegate: AnyObject {
+    func start(from cell: BookmarksFolderCell)
+
+    func showBookmarkDetail(for node: FxBookmarkNode, folder: FxBookmarkNode)
+}
+
+class BookmarksCoordinator: BaseCoordinator, BookmarksCoordinatorDelegate {
+    // MARK: - Properties
+
+    private let profile: Profile
+    private weak var parentCoordinator: LibraryCoordinatorDelegate?
+
+    // MARK: - Initializers
+
+    init(
+        router: Router,
+        profile: Profile,
+        parentCoordinator: LibraryCoordinatorDelegate?
+    ) {
+        self.profile = profile
+        self.parentCoordinator = parentCoordinator
+        super.init(router: router)
+    }
+
+    // MARK: - BookmarksCoordinatorDelegate
+
+    func start(from folderCell: BookmarksFolderCell) {
+        guard let folder = folderCell as? FxBookmarkNode else { return }
+        let viewModel = BookmarksPanelViewModel(profile: profile, bookmarkFolderGUID: folder.guid)
+        viewModel.bookmarkFolder = folder
```

Comment:
I don't think we should set the folder like this, it should be set from `reloadData` when the `BookmarksPanel` appears. If I remember well (it's been a while) the folder won't necessarily have the children nodes set, since we don't necessarily fetch sub-folders recursively. Also, if we set this `viewModel.bookmarkFolder = folder`, the `bookmarkFolder` will get overriden when we reloadData anyway.

Reply:
ah ok perfect, the i'm going to remove that line